spec/features/user_interacts_with_queue_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+feature 'User interacts with queue' do
+  scenario 'User adds and reorders videos in queue' do 
+    comedy = Fabricate(:category)
+    monk = Fabricate(:video, title: 'Monk', category: comedy)
+    south_park = Fabricate(:video, title: 'South Park', category: comedy)
+    futurama = Fabricate(:video, title: 'Futurama', category: comedy)
+
+    
+    sign_in
+
+    
+    add_video_to_queue(monk)
+    expect_video_added_to_queue(monk)
+    expect_add_my_queue_link_not_visible(monk)
+
+    add_video_to_queue(south_park)
+    expect_video_added_to_queue(south_park)
+    expect_add_my_queue_link_not_visible(south_park)
+
+    add_video_to_queue(futurama)
+    expect_video_added_to_queue(futurama)
+    expect_add_my_queue_link_not_visible(futurama)
+
+    visit queue_items_path
+    reorder_queue_positions(monk, 3)
+    reorder_queue_positions(south_park, 1)
+    reorder_queue_positions(futurama, 2)
+    
+    update_queue
```

Comment:
Updating queue would be also the 3 steps above. This means just clicking the update button.